app/models/task.rb

```diff
@@ -28,6 +28,8 @@ class Task < ActiveRecord::Base
              class_name: "Identity"
   belongs_to :assignable, polymorphic: true
 
+  belongs_to :procedure, ->{ joins(:tasks).where( tasks: { id: Task.where(assignable_type: 'Procedure' ) } ) }, foreign_key: :assignable_id
```

Comment:
I think
```
belongs_to :procedure, -> { joins(:tasks).where(tasks: { assignable_type: 'Procedure' }) }
```

Although doesn't this mean that a task belongs to multiple procedures? Should this be a `has_many` relationship instead?